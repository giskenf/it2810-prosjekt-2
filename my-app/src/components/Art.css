.pictures{
    display: flex;
    flex-direction: row;
}

@keyframes moveBird {
    from{
        transform: translate3d(250px,0,0);
    }
    0%{
        transform:translate3d(100,0,0);
    }
    100%{
        transform:translate3d(-1000px,0,0);
    }
}

#bird3{
    transform: translate(250px);
    animation: moveBird 8s linear infinite;
    animation-delay: 5s;
}
#bird2{
    transform: translate(250px);
    animation: moveBird 5s ease-in-out infinite;
    animation-delay: 2s;
}
#bird1{
    transform: translate(250px);
    animation: moveBird 7s linear infinite;
    animation-delay: 1s;
}


@keyframes colourChange {
    from{
        fill: #61dafb;
    }
    50%{
        fill: black;
    }
    100%{
        fill: #61dafb;
    }
}

#Frame1{
    fill: #61dafb;
    animation: colourChange 15s linear infinite;
    animation-delay: 3s;
}

@keyframes moveBubble {
    from{
        transform: translate3d(-100px,-250px,0);
    }
    0%{
        transform:translate3d(100px,0,-100px);
    }
    100%{
        transform:translate3d(0,-100%,500px);
    }
}


#Frame2{
    animation: colourChange 15s linear infinite;
    animation-delay: 3s;
}

.bubble{
    transform: translateY(250px);
    animation: moveBubble 5s linear infinite;
}

#bubble1{
    transform: translateY(250px);
    animation: moveBubble 10s linear infinite;
    animation-delay: 2s;
}
#bubble2{
    transform: translateY(250px);
    animation: moveBubble 3s linear infinite;
    animation-delay: 1s;
}
#bubble3{
    transform: translateY(250px);
    animation: moveBubble 6s linear infinite;
    animation-delay: 2s;
}

@keyframes rotateObject {
    0%{
        transform: rotate(0deg);
        transform-origin: center;
    }
    100%{
        transform: rotate(360deg);
        transform-origin: center;
    }
}

.sun{
    animation: rotateObject 15s linear infinite;
}

.planet{
    animation: rotateObject 10s linear infinite;
}
.planet2{
    animation: rotateObject 5s linear infinite;
}

