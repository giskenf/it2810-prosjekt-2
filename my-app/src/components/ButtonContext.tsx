import React, {useState, createContext, useContext, ReactNode} from 'react';



export const ButtonContext = createContext<any>({selectedOption:4});



export const ButtonProvider = (props: any) =>{

    const [selectedOption, setSelectedOption] = useState(ButtonContext)

    return(
        <ButtonProvider value={{selectedOption, setSelectedOption}}> {props.children} </ButtonProvider>
    )

}





/*export function useButtonContext(){
    return useContext(ButtonContext)
}

interface IupdateButtonContext{
    setOption: (option: Option) => void;
}

const UpdateButtonContext = createContext<IupdateButtonContext>({
    setOption: (option:Option) => {}
})

export function useUpdateButtonContext(){
    return useContext(UpdateButtonContext);
}
*/

/*export default function OptionProvider({children}:{children:ReactNode}){
    const [option,updateOption] = useState<Option>()

    const setOption = (o:Option) => {
        updateOption(o)
    };
    return(
        <ButtonContext.Provider value={option}>
            <UpdateButtonContext.Provider value={{setOption}}>
                {children}
            </UpdateButtonContext.Provider>
        </ButtonContext.Provider>

    )
}*/







