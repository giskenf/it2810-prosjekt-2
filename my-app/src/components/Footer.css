.BackgroundFooter{
    margin: 0;
    background-color: var(--sec-purple-color);
    padding: 0 20px;
    color: white;
    display: flex;
    flex-flow: row wrap;
    justify-content: space-around;
}
.BackgroundFooter#nightTheme{
    background-color: black;
}
.BackgroundFooter#jungleTheme{
    background-image: url("../svg/jungle_1.jpg");
}
.BackgroundFooter#dayTheme{
    background-color: #61dafb;
}

.radioButtonsContainer {
    display: flex;
    flex-wrap: wrap;
    justify-content: left;
}

.radioButtons {
    margin: 1%;
    display: grid;
    flex: 1;
    grid-template-rows: 1fr 1fr 1fr;
    align-self: stretch;
    justify-self: stretch;
    min-width: 100px;
    color: white;
    font-family: "Candara Light";
}

.buttons{
    font-family: 'Roboto', sans-serif;
    font-size: var(--font-tiny);
    background-color: var(--sec-orange-color);
    border: none;
    color: white;
    padding: 10px 10px 0 10px;
    margin: 3px 3px 3px 3px;
    height: 50px;
}


