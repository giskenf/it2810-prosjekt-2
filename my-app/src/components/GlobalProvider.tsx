import React, {useState, createContext, useContext, ReactNode, useMemo} from 'react';

export const GlobalContext = createContext<any>(1)

export const GlobalProvider = (props: any) =>{

    const [selectedOption, setSelectedOption] = useState(1)
    const [selectedTheme, setSelectedTheme] = useState(1)

    const buttonProvider = useMemo(() => ({
        selectedOption,setSelectedOption}),[selectedOption,setSelectedOption]);
    const themeProvider = useMemo(() => ({
        selectedTheme,setSelectedTheme}),[selectedTheme,setSelectedTheme]);

    return(
        <GlobalContext.Provider value={{buttonProvider,themeProvider}}> {props.children} </GlobalContext.Provider>
    )

}






/*export function useButtonContext(){
    return useContext(ButtonContext)
}

interface IupdateButtonContext{
    setOption: (option: Option) => void;
}

const UpdateButtonContext = createContext<IupdateButtonContext>({
    setOption: (option:Option) => {}
})

export function useUpdateButtonContext(){
    return useContext(UpdateButtonContext);
}
*/

/*export default function OptionProvider({children}:{children:ReactNode}){
    const [option,updateOption] = useState<Option>()

    const setOption = (o:Option) => {
        updateOption(o)
    };
    return(
        <ButtonContext.Provider value={option}>
            <UpdateButtonContext.Provider value={{setOption}}>
                {children}
            </UpdateButtonContext.Provider>
        </ButtonContext.Provider>

    )
}*/







