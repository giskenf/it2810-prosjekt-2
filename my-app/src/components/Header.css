.Background{
    margin: 0;
    background-color: var(--sec-purple-color);
    padding: 0 50px;
    color: white;
}

.NavArea{
    margin: 0;
    padding: 50px 0;
    display: flex;
    flex-direction: row;
}