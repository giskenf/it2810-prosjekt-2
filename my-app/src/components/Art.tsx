import React, {useState} from "react";
import "./Art.css";
import fox from '../svg/fox.svg'
import frame1 from '../svg/Dogforeste.svg'


export const Art = () => {
    const [id, setId] = useState(fox);

    return(
        <>
            <div className="Art">
                <img src={id} height="auto" width="auto"/>
            </div>
        </>
    );

};

