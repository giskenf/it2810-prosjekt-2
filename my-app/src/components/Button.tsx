import React, {useContext, useState} from "react";
import {GlobalContext} from "./GlobalProvider";

interface ButtonProps{
    changePoem:(a: number)=>void;
}
export function Button(props: ButtonProps){

    const {songProvider} = useContext(GlobalContext)

    return (
        <div className="radioButtonsContainer">
            <div className="radioButtons">
                <div>
                    <input type="radio" onChange={() => props.changePoem(2)}/>
                    Electronica
                </div>
                <div>
                    <input type="radio" onChange={() => props.changePoem(5)}/>
                    Piano
                </div>
                <div>
                    <input type="radio" onChange={() => songProvider.setSelectedSong('Rock')} checked={songProvider.selectedSong === "Rock"}/>
                    Rock
                </div>
                <button  onClick={()=>songProvider.setSelectedSong('Song')} >Reset </button>
            </div>
        </div>)


}