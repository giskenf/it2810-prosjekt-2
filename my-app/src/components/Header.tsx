import React, {useContext} from 'react';
import './Header.css'
import {TabsBar} from "./TabsBar";
import {GlobalContext} from "./GlobalProvider";
import {Store} from "./Store";



export const Header: React.FC = () => {
    const {themeProvider} = useContext(GlobalContext)
    return (
       <>
           <div className = "BackgroundHeader" id={themeProvider.selectedTheme}>
               <div className = "Headline">Museum</div>
               <TabsBar/>
               <Store/>
           </div>
       </>
);
};

