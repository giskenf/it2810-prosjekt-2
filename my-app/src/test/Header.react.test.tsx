
import React from 'react';
import renderer from 'react-test-renderer';
import {Header} from '../components/Header';
import {GlobalProvider} from "../components/GlobalProvider";

it('renders correctly', () => {
    const tree = renderer
        .create(<GlobalProvider><Header/></GlobalProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

