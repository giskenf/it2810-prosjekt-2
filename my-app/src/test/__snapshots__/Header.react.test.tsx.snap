// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Background"
>
  <div
    className="NavArea"
  >
    <a
      className="Logo"
    >
      Tittel
    </a>
    <div
      className="break"
    />
    <nav
      className="Nav"
    >
      <button
        className="NavButton"
        onClick={[Function]}
      >
        1 
      </button>
      <button
        className="NavButton"
        onClick={[Function]}
      >
        2 
      </button>
      <button
        className="NavButton"
        onClick={[Function]}
      >
        3 
      </button>
      <button
        className="NavButton"
        onClick={[Function]}
      >
        4 
      </button>
      <button
        className="NavButton"
        onClick={[Function]}
      >
        5 
      </button>
    </nav>
  </div>
</div>
`;
